### ISSUE TITLE
[NBug] Collection was modified; enumeration operation may not exec...

### ISSUE THREAD

**Issue by alvinsartor** [gitextensions/gitextensions#7527]
## Current behaviour

Crashed at opening.

## Expected behaviour

Not crashing.

## Steps to reproduce

1. Update to 3.3 (maybe this is the cause, maybe not)
2. Open Git Extensions
3. Select a repository.
4. Crashed

## Error Details
```
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   at System.Linq.LinqExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action)
   at GitUI.PluginRegistry.Register(IGitUICommands gitUiCommands)
   at GitUI.CommandsDialogs.FormBrowse.RegisterPlugins()
   at GitUI.CommandsDialogs.FormBrowse.SetGitModule(Object sender, GitModuleEventArgs e)
   at GitUI.CommandsDialogs.BrowseDialog.DashboardControl.UserRepositoriesList.listView1_MouseClick(Object sender, MouseEventArgs e)
   at System.Windows.Forms.Control.OnMouseClick(MouseEventArgs e)
   at System.Windows.Forms.ListView.WmReflectNotify(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at GitUI.UserControls.ExListView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

## Additional information
GE crashed at opening, just after selecting the repository.
I just installed the new version (3.3), so that might have been involved in the crash.

Similar to #7117 but the stack trace is different.

It crashed only once, the following times everything went on normally.

## Environment
- Git Extensions 3.3.0.7719
- Build e52f5178e8d3c9899efbe6fae66026865afa3fe1
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4075.0
- DPI 120dpi (125% scaling)

**Comment by alvinsartor** [comment]
Duplicate of #6924

**Comment by gerhardol** [comment]
@pmiossec Did you do a change related to this in 3.3?

**Comment by pmiossec** [comment]
No, nothing that I can remember. It should be a rare race condition already there.

But we could easily add some `lock(){}` as there is already one at one place...
